tracer/src/Datadog.Trace.Tools.Runner/GacUninstallCommand.cs

```diff
@@ -39,34 +39,48 @@ private void Execute(InvocationContext context)
             return;
         }
 
-        using var container = NativeMethods.CreateAssemblyCache();
-        var hr = container.AssemblyCache.UninstallAssembly(UninstallAssemblyFlags.None, assemblyName!, IntPtr.Zero, out var position);
+        using var gacMethods = GacNativeMethods.Create();
+        var assemblyCache = gacMethods.CreateAssemblyCache();
 
-        switch (position)
+        var result = Hresult.S_OK;
+        var installedAssemblyNames = gacMethods.GetAssemblyNames(assemblyName!);
+        if (installedAssemblyNames.Count == 0)
         {
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_ALREADY_UNINSTALLED:
-                Utils.WriteWarning($"Assembly '{assemblyName}' was already uninstalled from the GAC.");
-                context.ExitCode = hr;
-                break;
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_REFERENCE_NOT_FOUND:
-                Utils.WriteWarning($"Assembly '{assemblyName}' not found in the GAC.");
-                context.ExitCode = hr;
-                break;
-            case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_STILL_IN_USE:
-                Utils.WriteWarning($"Assembly '{assemblyName}' is still in use.");
-                context.ExitCode = hr;
-                break;
+            Utils.WriteWarning($"Assembly '{assemblyName}' is not installed the GAC.");
+            return;
+        }
+
+        foreach (var installedAssemblyName in installedAssemblyNames)
+        {
+            var hr = assemblyCache.UninstallAssembly(UninstallAssemblyFlags.None, installedAssemblyName.Name, IntPtr.Zero, out var position);
+            switch (position)
+            {
+                case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_ALREADY_UNINSTALLED:
+                    Utils.WriteWarning($"Assembly '{installedAssemblyName.FullName}' was already uninstalled from the GAC.");
+                    break;
+                case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_REFERENCE_NOT_FOUND:
+                    Utils.WriteWarning($"Assembly '{installedAssemblyName.FullName}' not found in the GAC.");
+                    break;
+                case UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_STILL_IN_USE:
```

Comment:
You _may_ want to handle `IASSEMBLYCACHE_UNINSTALL_DISPOSITION_HAS_INSTALL_REFERENCES` too - I had to in the `FleetInstaller` project (where we are duplicating a lot of this logic - we should perhaps consider merging those in the future

Reply:
done in [changes from review](https://github.com/DataDog/dd-trace-dotnet/pull/6995/commits/ce9ad222a7cb1f9caafb21b12a5c01e11b10b06c)